test/api/utils.ts

```diff
@@ -0,0 +1,87 @@
+import _ from 'lodash'
+import { RippleAPI } from 'ripple-api'
+import assert from 'assert-diff'
+const { schemaValidator } = RippleAPI._PRIVATE
+
+export type TestFn = (
```

Comment:
How does this typing work? Some of your tests take in a single `typescript` argument. I would think that typescript would yell at you that you're missing the `address` parameter.

Should we mark `address: string` as optional `address?: string?`

Reply:
By `typescript` argument do you mean the `api` argument? If so, you can think of it like this:

1. We tell TypeScript that this kind function is called with `api` & `address` arguments.
2. When we implement this kind of function, we can choose to ignore arguments if we don't need them. They'll still get passed, and you couldn't expect a *different* kind of argument (a number vs. a string) but ignoring them is fine.

Reply:
I understand that javascript can ignore arguments. According to this though, you'd need to mark an argument as optional to get it to compile: https://www.typescriptlang.org/docs/handbook/functions.html#optional-and-default-parameters

Reply:
That's right, but It only works in the other direction. Think of it this way:

- I am a function that takes `(a, b, c)`. If you pass me a function of type `(a, b)` I will get mad, since you're not giving me the `c` arg I expected.
- I am a function that takes `(a, b, c)`. If you pass me a function of type `(a, b, c, d)` that is fine, since you're passing me all the args that I ask for. You're passing me a `d` arg that I didn't ask for, but since you matched my `a,b,c` type signature I don't care enough to throw an error.